# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/rafiaburdibaloch-ui/flyrank-rafia/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

### My lane: Refresh / Content Opportunity Scoring

I chose the Refresh / Content Opportunity Scoring lane because I want to investigate which content pages should be reviewed first for improvement, refresh, protection, pruning, or monitoring. This lane connects the available content and search-performance data to a practical decision: helping a reviewer prioritize limited time and attention. I am interested in this problem because the dataset contains observable signals such as impressions, clicks, position, sessions, content age, and freshness that may help identify pages with meaningful opportunities. I will start with the starter dataset and use the evidence to refine my question as I learn more about the data.


In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/rafiaburdibaloch-ui/flyrank-rafia"   # <-- your own repo
REPO_DIR = "flyrank-rafia"                                     # <-- change to match your repo name

if IN_COLAB:
    if not os.path.isdir(REPO_DIR):
        subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
    os.chdir(REPO_DIR)

print("Working dir:", os.getcwd())
print(os.listdir("."))

Working dir: /flyrank-rafia/flyrank-rafia
['skills', 'AGENTS.md', '.gitignore', 'docs', 'data', 'GUIDE.md', 'submission', 'outputs', 'LICENSE', '.git', '.github', 'README.md', '01_first_look_and_discovery.ipynb', 'notebooks', 'CLAUDE.md', 'SETUP.md', 'requirements.txt', 'DATA_USE.md', 'scripts', 'work']


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

### My question

**Research question**: Which content pages should be reviewed first for a possible refresh or other appropriate action, based on their observed search and engagement signals?

**Decision:** A content reviewer needs to decide which pages should receive attention first when time and resources are limited.

**Action:** The reviewer could prioritize a page for refresh, improvement, protection, pruning, or continued monitoring based on the available evidence.

**Cost of a wrong call:** If a page is incorrectly prioritized, time and resources may be spent on a page that has little opportunity. If a useful page is missed, a potentially valuable improvement opportunity may be overlooked.

**Why data or ML could help:** The dataset contains multiple observable signals, such as impressions, clicks, position, sessions, content age, and freshness. Looking at these signals together may help identify patterns and prioritize pages more consistently than looking at one metric at a time.


In [18]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [19]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of rows:", len(df))
print("Number of columns:", len(df.columns))
print("Number of unique content pages:", df["content_id"].nunique())

declining_rate = (df["trend_direction"].str.lower() == "down").mean()
print(f"\nShare of pages currently declining: {declining_rate:.1%}")

stale_pages = (df["days_since_last_update"] >= 180).mean()
print(f"Share of pages stale (not updated in 180+ days): {stale_pages:.1%}")

stale_and_visible = ((df["days_since_last_update"] >= 180) & (df["impressions_90d"] >= 500)).sum()
print(f"Pages that are BOTH stale AND still getting real traffic: {stale_and_visible}")

Number of rows: 30000
Number of columns: 44
Number of unique content pages: 30000

Share of pages currently declining: 54.2%
Share of pages stale (not updated in 180+ days): 0.6%
Pages that are BOTH stale AND still getting real traffic: 17


The starter dataset contains 30,000 rows, 44 columns, and 30,000 unique content pages — one row per page, no duplicates. Over half (54.2%) of all pages are currently trending down, showing decline is common here, not an edge case — a real, sizeable prioritization problem. Only 0.6% of pages are stale (180+ days since update), meaning most content is fairly fresh, so staleness alone will be a weak signal in this dataset. Of that small stale group, 17 pages are both stale and still getting real traffic (500+ impressions in 90 days) — a concrete example of the kind of overlooked opportunity this lane is meant to catch.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What this work can say: This project can surface pages where observed signals — such as declining engagement, staleness, or high visibility with weak performance — suggest they may be worth a reviewer's attention. Findings will be described as observed and directional, based on patterns in this specific anonymized dataset. Any prioritization score is meant as decision-support: a starting point for a human reviewer, not a final verdict.

What this work cannot say: This project cannot claim any signal causes a page to decline, and cannot predict how Google's ranking algorithm works or will behave. It cannot guarantee a "top priority" page will actually improve if refreshed — only that it resembles pages that historically needed attention. It cannot generalize beyond this dataset to real-world SEO or content strategy broadly.

In [20]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
# Quick sanity check backing Section 4: confirm we're not accidentally using
# any outcome-derived or leaky columns as "evidence" in our claims
leaky_terms = ["trend_pct", "health_score", "needs_"]
present_leaky = [c for c in df.columns if any(t in c for t in leaky_terms)]
print("Columns that must be excluded from any future scoring model (leakage risk):")
print(present_leaky)

Columns that must be excluded from any future scoring model (leakage risk):
['trend_pct']


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.